src/main/java/com/aws/greengrass/config/Topic.java

```diff
@@ -312,6 +312,9 @@ public synchronized Topic dflt(Object dflt) {
     public boolean equals(Object o) {
         if (o instanceof Topic) {
             Topic t = (Topic) o;
+            if (getOnce() != null && t.getOnce() != null) {
+                return getName().equals(t.getName()) && getOnce().equals(t.getOnce());
+            }
```

Comment:
is this change safe? should you create a new method for this?

Reply:
I don't know for sure if this is safe, I don't think creating a new method will help simplify things because we need to compare the values, not just the names.

Reply:
Please do not modify core logic like this.